### PR TITLE
Feature/integrations redux store

### PR DESCRIPTION
**Pull request by deeleman**
This introduces NGRX-powered state management on Integrations, required for adding metrics later on and properly persisting integrations throughout the application.

Please check the README.md file

### CODE REVIEW

**Comment by kahboom** [comment]
Great work so far!

File: app/ui/src/app/app.component.ts [hunk 1]
```diff
@@ -12,9 +12,10 @@ import { saveAs } from 'file-saver';
 import { Restangular } from 'ngx-restangular';
 import { Notification, NotificationEvent } from 'patternfly-ng';
 import { Observable } from 'rxjs/Observable';
-import { ModalService } from '@syndesis/ui/common/modal/modal.service';
-import { NavigationService } from '@syndesis/ui/common/navigation.service';
-import { UserService, User } from '@syndesis/ui/platform';
+import { Store } from '@ngrx/store';
+import { UserService, User, PlatformActions, PlatformState } from '@syndesis/ui/platform';
+import { ModalService } from './common/modal/modal.service';
```
Comment (seanforyou23) [inline app/ui/src/app/app.component.ts]: Can we continue using the "@syndesis/ui/" path alias here?

File: app/ui/src/app/app.module.ts [hunk 2]
```diff
@@ -20,8 +20,8 @@ import { AppComponent } from './app.component';
 import { AppRoutingModule } from './app.routing';
 import { SyndesisCommonModule } from './common';
 import { appConfigInitializer, ConfigService } from './config.service';
-import { StoreModule as LegacyStoreModule } from '@syndesis/ui/store/store.module';
-import { platformReducer, SYNDESIS_GUARDS } from './platform';
+import { StoreModule as LegacyStoreModule } from './store/store.module';
+import { platformReducer, PlatformEffects, platformEndpoints, SYNDESIS_GUARDS } from './platform';
```
Comment (seanforyou23) [inline app/ui/src/app/app.module.ts]: Same question regarding utilizing "@syndesis/ui" path alias

File: app/ui/src/app/app.component.ts (hunk 1 above)
Comment (deeleman) [reply app/ui/src/app/app.component.ts]: Root elements (app.component.ts, app.module.ts, app.etc.ts) usually refer to the folders themselves instead. It’s a good way to inform hierarchies visually.

Note to self: include modal and all its code entities as part of the main common barrel.

File: app/ui/src/app/app.module.ts (hunk 2 above)
Comment (deeleman) [reply app/ui/src/app/app.module.ts]: As you can see I removed those on purpose. Let’s stick to relative paths when consuming tokens from root classes.

**Pull request merged**